File: code-samples/messaging/quick-start.php
<?php
// Remember to modify the path to where you installed the RingCentral SDK and saved your .env file!
require('./../vendor/autoload.php');
$dotenv = Dotenv\Dotenv::createImmutable(__DIR__ . '/../');
$dotenv->load();

// For the purpose of testing the code, we put the SMS recipient number in the environment variable.
// Feel free to set the SMS recipient directly.
$RECIPIENT    = $_ENV['SMS_RECIPIENT'];

# Instantiate the SDK and get the platform instance
$rcsdk = new RingCentral\SDK\SDK( $_ENV['RC_CLIENT_ID'],
                                  $_ENV['RC_CLIENT_SECRET'],
                                  $_ENV['RC_SERVER_URL'] );
$platform = $rcsdk->platform();

// Authenticate a user using a personal JWT token
try {
  $platform->login( [ "jwt" => $_ENV['RC_JWT'] ] );
  read_extension_phone_number_detect_sms_feature();
} catch (\RingCentral\SDK\Http\ApiException $e) {
  exit("Unable to authenticate to platform. Check credentials. " . $e->message . PHP_EOL);
}

/*
  Read phone number(s) that belongs to the authenticated user and detect if a phone number
  has the SMS capability
*/
function read_extension_phone_number_detect_sms_feature(){
  global $platform;
  $endpoint = "/restapi/v1.0/account/~/extension/~/phone-number";
  $resp = $platform->get($endpoint);
  $jsonObj = $resp->json();
  foreach ($resp->json()->records as $record){
    foreach ($record->features as $feature){
      if ($feature == "SmsSender"){
        // If a user has multiple phone numbers, check and decide which number
        // to be used for sending SMS message.
        return send_sms($record->phoneNumber);
      }
    }
  }
  if (count($jsonObj->records) == 0){
    exit("This user does not own a phone number!");
  }else{
    exit("None of this user's phone number(s) has the SMS capability!");
  }
}

/*
 Send a text message from a user own phone number to a recipient number
*/
function send_sms($fromNumber){
  global $platform, $RECIPIENT;
  try {
    $requestBody = array(
       'from' => array ('phoneNumber' => $fromNumber),
       'to' => array( array('phoneNumber' => $RECIPIENT) ),
       // To send group messaging, add more (max 10 recipients) 'phoneNumber' object. E.g.
       /*
       'to' => array(
          array('phoneNumber' => $RECIPIENT),
          array('phoneNumber' => 'Recipient-Phone-Number')
        ),
       */
       'text' => 'Hello World!'
    );
    $endpoint = "/account/~/extension/~/sms";
    $resp = $platform->post($endpoint, $requestBody);
    $jsonObj = $resp->json();
    print("SMS sent. Message id: " . $jsonObj->id . PHP_EOL);
    check_message_status($jsonObj->id);
  } catch (\RingCentral\SDK\Http\ApiException $e) {
    exit("Error message: " . $e->message . PHP_EOL);
  }
}

/*
 Check the sending message status until it's out of the queued status
*/
function check_message_status($messageId){
    global $platform;
    try {
        $endpoint = "/restapi/v1.0/account/~/extension/~/message-store/".$messageId;
        $resp = $platform->get($endpoint);
        $jsonObj = $resp->json();
        print("Message status: " . $jsonObj->messageStatus . PHP_EOL);
        if ($jsonObj->messageStatus == "Queued"){
          sleep(2);
          check_message_status($jsonObj->id);
        }
    } catch (\RingCentral\SDK\Http\ApiException $e) {
      exit("Error message: " . $e->message . PHP_EOL);
    }
}
?>


<?php
/**********************************************************
 Code snippet section for boostrap testing purpose
**********************************************************/
$RECIPIENT2    = $_ENV['SMS_RECIPIENT2'];
boostrap_test_function();
function boostrap_test_function(){
  /*
  print_r ("Test reading number features". PHP_EOL);
  sleep(2);
  include_once (__DIR__ .'/code-snippets/number-features.php');

  return
  print_r ("Test sending MMS". PHP_EOL);
  sleep(2);
  include_once (__DIR__ .'/code-snippets/send-mms.php');

  print_r ("Test sending Fax". PHP_EOL);
  sleep(2);
  include_once (__DIR__ .'/code-snippets/send-fax.php');

  print_r ("Test reading message store". PHP_EOL);
  sleep(2);
  include_once (__DIR__ .'/code-snippets/message-store.php');

  print_r ("Test export message store". PHP_EOL);
  sleep(2);
  include_once (__DIR__ .'/code-snippets/message-store-export.php');
  */

  print_r ("Test sending a2p SMS". PHP_EOL);
  sleep(2);
  include_once (__DIR__ .'/code-snippets/send-a2p-sms.php');
}
?>

File: code-samples/messaging/quick-start.rb
require 'ringcentral'
require 'dotenv'
# Remember to modify the path to where you saved your .env file!
Dotenv.load("./../.env")

# For the purpose of testing the code, we put the SMS recipient number in the environment variable.
# Feel free to set the SMS recipient directly.
RECIPIENT    = ENV['SMS_RECIPIENT']

# Read phone number(s) that belongs to the authenticated user and detect if a phone number
# has the SMS capability
def read_extension_phone_number_detect_sms_feature()
  begin
    endpoint = "/restapi/v1.0/account/~/extension/~/phone-number"
    resp = $platform.get(endpoint)
    for record in resp.body['records'] do
      for feature in record['features'] do
        if feature == "SmsSender"
          # If user has multiple phone numbers, check and decide which number
          # to be used for sending SMS message.
          return send_sms(record['phoneNumber'])
        end
      end
    end
    if resp.body['records'].length == 0
      puts ("This user does not own a phone number!")
    else
      puts("None of this user's phone number(s) has the SMS capability!")
    end
  rescue StandardError => e
    puts (e)
  end
end

# Send a text message from a user own phone number to a recipient number
def send_sms(fromNumber)
  begin
    bodyParams = {
          from: { phoneNumber: fromNumber },
          to: [{ phoneNumber: RECIPIENT }],
          # To send group messaging, add more (max 10 recipients) 'phoneNumber' object. E.g.
          #
          #to: [
          #       { phoneNumber: RECIPIENT },
          #       { phoneNumber: 'Recipient-Phone-Number' }
          #     ],
          text: 'Hello World!'
      }
    endpoint =  "/restapi/v1.0/account/~/extension/~/sms"
    resp = $platform.post(endpoint, payload: bodyParams)
    puts ("SMS sent. Message id: " + resp.body['id'].to_s)
    check_message_status(resp.body['id'])
  rescue StandardError => e
    puts (e)
  end
end

# Check the sending message status until it's out of the queued status
def check_message_status(messageId)
  begin
    endpoint = "/restapi/v1.0/account/~/extension/~/message-store/" + messageId.to_s
    resp = $platform.get(endpoint)
    puts ("Message status: " + resp.body['messageStatus'])
    if (resp.body['messageStatus'] == "Queued")
      sleep(2)
      check_message_status(resp.body['id'])
    end
  rescue StandardError => e
    puts (e)
  end
end

# Instantiate the SDK and get the platform instance
$platform = RingCentral.new( ENV['RC_CLIENT_ID'], ENV['RC_CLIENT_SECRET'], ENV['RC_SERVER_URL'] )

# Authenticate a user using a personal JWT token
def login()
  begin
    $platform.authorize(jwt: ENV['RC_JWT'])
    read_extension_phone_number_detect_sms_feature()
  rescue StandardError => e
    puts ("Unable to authenticate to platform. Check credentials." + e.to_s)
  end
end

login()
##############################
# End of quick start section
##############################


########
# Code snippet section for boostrap testing purpose
########
def boostrap_test_function()

    # sleep(2)
    # puts "Test sending mms"
    # require_relative './code-snippets/send-mms'
    # read_extension_phone_number_detect_mms_feature()
    # return
    # sleep(2)
    # puts "Test sending Fax"
    # require_relative './code-snippets/send-fax'
    # send_fax()
    #
    # sleep(2)
    # puts "Test reading message store"
    # require_relative './code-snippets/message-store'
    # read_extension_message_store()
    #
    # sleep(2)
    # puts "Test reading number features"
    # require_relative './code-snippets/number-features'
    # detect_sms_feature()

    # sleep(2)
    # puts "Test export message store"
    # require_relative './code-snippets/message-store-export'
    # create_message_store_report()

    sleep(2)
    puts "Test sending batch sms"
    require_relative './code-snippets/send-a2p-sms'
    read_extension_phone_number_detect_a2psms_feature()
end

boostrap_test_function()
